golang/arguments.go

```diff
@@ -0,0 +1,103 @@
+package tchannel
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+)
+
+// An Input is able to read an argument from a call body
```

Comment:
Suggest making comments golint-compatible (e.g. start the comment with the type name you're describing):

```
// Input describes a type...
```